txpool/pool.go

```diff
@@ -613,22 +613,22 @@ func (p *TxPool) AddNewGoodPeer(peerID types.PeerID) { p.recentlyConnectedPeers.
 func (p *TxPool) Started() bool                      { return p.started.Load() }
 
 func (p *TxPool) best(n uint16, txs *types.TxsRlp, tx kv.Tx, onTopOf, availableGas uint64, toSkip mapset.Set[[32]byte]) (bool, int, error) {
+	p.lock.Lock()
```

Comment:
I think we have un-mentioned convention: 
methods from small letter - are thread-unsafe, methods from capital letter manage mutexes. 

Otherwise can get double-locking problem.